TenancyInformationApi.Tests/V1/Controllers/TenancyInformationControllerTests.cs

```diff
@@ -0,0 +1,46 @@
+using TenancyInformationApi.V1.Controllers;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc;
+using NUnit.Framework;
+using TenancyInformationApi.V1.UseCase.Interfaces;
+using Moq;
+using TenancyInformationApi.V1.Boundary.Response;
+
+namespace TenancyInformationApi.Tests.V1.Controllers
+{
+    [TestFixture]
+    public class TenancyInformationControllerTests
+    {
+        private TenancyInformationController _classUnderTest;
+        private Mock<IGetTenancyByIdUseCase> _getByIdMock;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _getByIdMock = new Mock<IGetTenancyByIdUseCase>();
+            _classUnderTest = new TenancyInformationController(_getByIdMock.Object);
+        }
+
+        [Test]
+        public void ViewRecordReturnsResponseWithStatus()
+        {
+            ObjectResult response;
+
+            // 1231 is not a valid tag_ref - should return BadRequest400
+            response = _classUnderTest.ViewRecord("1231") as ObjectResult;
+            response.Should().NotBeNull();
+            response?.StatusCode.Should().Be(400);
+
+            // 123/1 is not in the db - should return NotFound400
+            response = _classUnderTest.ViewRecord("123-1") as ObjectResult;
+            response.Should().NotBeNull();
+            response?.StatusCode.Should().Be(404);
+
+            // 123/1 has been added - should return OK200
+            _getByIdMock.Setup(x => x.Execute("123/1")).Returns(new TenancyInformationResponse());
```

Comment:
I think your check on the controller (Line 45) sees if there's a tenancy ref and returns 404 if there's not. So you'll need to setup your returned object here with a tenancy ref as a minimum.